### PR TITLE
Endurance tests

### LINKED ISSUE #12: Load testing
10K Messages for Rq/Rep per second

### LINKED ISSUE #50: Use lots of Patterns simultaneously
With a single Socket... (PubSub and RqRep on a single socket)
With different Sockets...(PubSub on 2 Sockets, same address, different address etc.)

### PR DESCRIPTION
**Pull request by TimoHeiten**
Added new Project for manual load testing.

Is just a scaffold for now, yet it already revealed some issues when the library is running with load (e.g. watermarks for pub sub and multithreading issues when using a single socket (underlying NetMQ lib is not threadsafe here)

### CODE REVIEW

File: src/heitech.zer0mqXt.core/utils/PollerUtils.cs [hunk 1]
```diff
@@ -10,10 +11,9 @@ public static void DisposeOf(this NetMQPoller poller, NetMQSocket anySocket, Eve
             anySocket.ReceiveReady -= receiveDelegate;
             if (poller != null && poller.IsRunning)
             {
-                poller.Stop();
-                poller.Remove(anySocket);
-                poller.Dispose();
-                anySocket.Dispose();
+                // needs to be done from the Poller´s thread or else a race condition is encountered.
+                new Task(() => { poller.Remove(anySocket); anySocket.Dispose(); })
+                .Start(poller);
```
Comment (Flash0ver) [inline src/heitech.zer0mqXt.core/utils/PollerUtils.cs:16]: Cool - I didn't know about the `TaskScheduler` overload of `Task.Start`.

File: tests/zeromq.endurance/Program.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+            await Task.WhenAll(tasks);
+
+            var tasks2 = new List<Task> {
+                tester.SimpleMessagesRaw(chunkMessages(2)),
+                tester.SimpleMessagesRaw(chunkMessages(4)),
+                tester.SimpleMessagesRaw(chunkMessages(8)),
+                tester.SimpleMessagesRaw(chunkMessages(16)),
+                tester.SimpleMessagesRaw(chunkMessages(16))
+            };
```
Comment (Flash0ver) [inline tests/zeromq.endurance/Program.cs:60]: Are you intentionally not awaiting this set of Tasks?

File: src/heitech.zer0mqXt.core/utils/PollerUtils.cs (hunk 1 above)
Comment (TimoHeiten) [reply src/heitech.zer0mqXt.core/utils/PollerUtils.cs:16]: Yeah that one is quite useful for things with thread affinity like the Poller for example.

File: tests/zeromq.endurance/Program.cs (hunk 2 above)
Comment (TimoHeiten) [reply tests/zeromq.endurance/Program.cs:60]: 9. I just forgot :D thank you. That also explains some of the issues i had during development...

File: tests/zeromq.endurance/Program.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+            await Task.WhenAll(tasks);
```
Comment (Flash0ver) [inline tests/zeromq.endurance/Program.cs]: Did you mean to await `tasks2` here, rather than `tasks` again?

Comment (TimoHeiten) [reply tests/zeromq.endurance/Program.cs]: dang it xD

**Review by Flash0ver — APPROVED** [review]

**Pull request merged**